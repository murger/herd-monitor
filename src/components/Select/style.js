import styled from 'styled-components';

export const Wrapper = styled.div`
  position: relative;
  display: inline-block;
  border-radius: 8px;
  vertical-align: baseline;
  background-color: #4629d3;
  cursor: pointer;

  &:hover {
    background-color: #295cd3;
  }

  &:after {
    z-index: 1;
    content: "";
    position: absolute;
    top: 50%;
    right: 8px;
    border-left: 6px solid transparent;
    border-right: 6px solid transparent;
    border-top: 7px solid rgba(255, 255, 255, 0.5);
    pointer-events: none;
    transform: translateY(-50%);
  }
`;

export const Selector = styled.select`
  display: block;
  font-size: 16px;
  font-weight: 500;
  font-family: Mabry;
  color: white;
  line-height: 1;
  padding: 10px 0;
  padding-left: 12px;
  padding-right: 28px;
  width: 100%;
  box-sizing: border-box;
  margin: 0px;
  border: 0px;
  border-radius: 0px;
  appearance: none;
  background-color: transparent;
  outline: none;

  &::-ms-expand {
    display: none;
  }

  option {
    font-weight: normal;
  }
`;
