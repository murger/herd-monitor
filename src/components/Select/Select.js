import React, { Fragment } from 'react';
import { Wrapper, Selector } from './style';

const Select = (props) => {
	return (
    <Fragment>
      <Wrapper>
        <Selector>
          {props.children}
        </Selector>
      </Wrapper>
    </Fragment>
	);
};

export default Select;
