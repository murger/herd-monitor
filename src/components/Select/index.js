import React from 'react';
import { Wrapper, Selector } from './style';

const Select = (props) => {
	return (
		<Wrapper>
			<Selector>
				{props.children}
			</Selector>
		</Wrapper>
	);
};

export default Select;
