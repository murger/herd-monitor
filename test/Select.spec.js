import React from 'react';
import renderer from 'react-test-renderer';
import Select from '../src/components/Select';

it('renders the correctly', () => {
  const tree = renderer.create(<Select />).toJSON();
  expect(tree).toMatchSnapshot();
});
