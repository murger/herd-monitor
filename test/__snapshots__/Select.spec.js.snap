// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the correctly 1`] = `
<div
  className="sc-bdVaJa fRmVhe"
>
  <select
    className="sc-bwzfXH ejOUFP"
  />
</div>
`;
